### ISSUE TITLE
Attempt to access the step user before it was initialised. 

### ISSUE THREAD

**Issue by sbt-sysadmin** [cbluesprl/moodle-quiz_export#12]
Moodle 4.3.4+ (Build: 20240530) | main moodle-quiz_export branch.

Able to export one Quiz attempt at a time, but have to click on Show report button before each export of a single Quiz attempt.

When attempting to export multiple Quiz attempts, I am presented with a confirmation window asking if I want to export selected attempts.

When I click on Yes, I see this error message:

Attempt to access the step user before it was initialised. Did you forget to call question_usage_by_activity::preload_all_step_users() or similar?

line 204 of /question/engine/questionattemptstep.php: call to debugging()
line 215 of /question/engine/questionattemptstep.php: call to question_attempt_step->get_user()
line 524 of /question/engine/renderer.php: call to question_attempt_step->get_user_fullname()
line 495 of /question/engine/renderer.php: call to core_question_renderer->action_author()
line 120 of /question/engine/renderer.php: call to core_question_renderer->response_history()
line 113 of /question/behaviour/behaviourbase.php: call to core_question_renderer->question()
line 912 of /question/engine/questionattempt.php: call to question_behaviour->render()
line 483 of /question/engine/questionusage.php: call to question_attempt->render()
line 1345 of /mod/quiz/classes/quiz_attempt.php: call to question_usage_by_activity->render_question()
line 1307 of /mod/quiz/classes/quiz_attempt.php: call to mod_quiz\quiz_attempt->render_question_helper()
line 199 of /mod/quiz/classes/output/renderer.php: call to mod_quiz\quiz_attempt->render_question()
line 67 of /mod/quiz/classes/output/renderer.php: call to mod_quiz\output\renderer->questions()
line 314 of /mod/quiz/report/export/export.php: call to mod_quiz\output\renderer->review_page()
line 270 of /mod/quiz/report/export/export.php: call to quiz_export_engine->render()
line 250 of /mod/quiz/report/export/export.php: call to quiz_export_engine->get_review_html()
line 107 of /mod/quiz/report/export/export.php: call to quiz_export_engine->all_questions()
line 198 of /mod/quiz/report/export/report.php: call to quiz_export_engine->a2pdf()
line 164 of /mod/quiz/report/export/report.php: call to quiz_export_report->export_attempts()
line 86 of /mod/quiz/report/export/report.php: call to quiz_export_report->process_actions()
line 83 of /mod/quiz/report.php: call to quiz_export_report->display()

-=-

**Comment by shenrard** [comment]
Hi there,

Thank you for reporting this issue. I tried to reproduce the problem using a simple quiz with a true/false question, but I wasn't able to encounter the same issue during export.

Could you please provide more details about the quiz you're working with? For example:

- The type and number of questions used.
- Any specific settings you applied to the quiz.
- The exact steps you're taking when you encounter the issue.
- Any error messages or logs that could help us identify the problem.

Apologies for the inconvenience, and we'll do our best to resolve this as quickly as possible.

Thank you!

**Comment by sbt-sysadmin** [comment]
1) 20 Short-Answer Questions. Have tried various courses with different numbers of questions.

2) I have all Quiz settings set to default.

3a) Click on checkmark next to usernames under Quiz - Results - Quiz exporting.
3b) Scroll down to bottom and click on Export selected attempts.
3c) When Confirmation dialog (Export selected attempts?) appears, click on Yes.

4) I have checked various /var/log files, but have found nothing related to the error I detailed in original issue submission.

Thank you.
